infrastructure/lib/constructs/compliant-resources.ts

```diff
@@ -64,3 +65,53 @@ export class SecureTable extends cdk.Construct {
     this.table = table;
   }
 }
+
+export interface SecureRestApiProps extends apigw.RestApiBaseProps {
+  /**
+   * The properties to configure the API Gateway
+   */
+  apiDefinition?: apigw.ApiDefinition;
+}
+
+/**
+ * Creates a secure API Gateway with properties enabled for compliance
+ *  - cache enabled and encrypted by default
+ *  - execution logs enabled by default
+ */
+export class SecureRestApi extends cdk.Construct {
+  readonly restApi: apigw.RestApiBase;
+
+  constructor(scope: cdk.Construct, id: string, props: SecureRestApiProps) {
+    super(scope, id);
+    const baseProps = {
+      // This is slightly repetitive between endpointTypes and parameters.endpointConfigurationTypes; however, due
+      // to underlying CloudFormation behaviors, endpointTypes is not evaluated entirely correctly when a
+      // parameter specification is given. Specifying both ensures that we truly have a regional endpoint rather than
+      // edge.
+      endpointTypes: [apigw.EndpointType.REGIONAL],
+      parameters: {
+        endpointConfigurationTypes: apigw.EndpointType.REGIONAL,
+      },
```

Comment:
Future TODO: only add this block for the `SpecRestApi` case since that's the only time it's actually needed.

Reply:
Okay, I'll make a note. and make sure to update. Thanks!